dask/array/core.py

```diff
@@ -2852,10 +2838,25 @@ def choose(a, choices):
 """.strip()
 
 
+@wraps(np.argwhere)
+def argwhere(a):
+    from .creation import indices
+
+    nz = (a != 0).flatten()
```

Comment:
Have added support for different non-zero cases in a function called `isnonzero`. This handles numeric values, strings, and has a miscellaneous fallback case (e.g. `object`s). Also have added tests to exercise these cases.